backend/scheduler/src/jobs/resolve-sports-markets.ts

```diff
@@ -0,0 +1,120 @@
+import { createSupabaseDirectClient } from 'shared/supabase/init'
+import { log } from 'shared/utils'
+import { Contract, CPMMMultiContract } from 'common/contract'
+import { HOUSE_LIQUIDITY_PROVIDER_ID } from 'common/antes'
+import { resolveMarketHelper } from 'shared/resolve-market-helpers'
+import { SportsGames } from 'common/sports-info'
+import { api } from 'web/lib/api/api'
+
+export async function resolveSportsMarkets() {
+  const pg = createSupabaseDirectClient()
+
+  try {
+    const data = await api('get-completed-sports-games', {})
+    const completedGames = (data.schedule || []) as SportsGames[]
+    log(`Processing ${completedGames.length} completed games.`)
+
+    const unresolvedContracts = await pg.map<Contract>(
+      `
+      SELECT *
+      FROM contracts
+      WHERE resolution IS NULL
+        AND data->>'sportsEventId' IS NOT NULL
+      `,
+      [],
+      (row) => row as Contract
+    )
+
+    for (const game of completedGames) {
+      const matchingContracts = unresolvedContracts.filter(
+        (c) => c.sportsEventId === game.idEvent
+      )
+      if (matchingContracts.length === 0) continue
+
+      log(
+        `Found ${matchingContracts.length} contracts for game ${game.idEvent}`
+      )
+
+      const resolver = await pg.one('SELECT * FROM users WHERE id = $1', [
```

Comment:
`pg.one` is going to throw an error if the row is not found, so there's no reason to add a !resolver check.